cami.c

```diff
@@ -511,15 +574,19 @@ void ami_set_debug(struct ami_session *ami, int fd)
 
 int ami_set_debug_level(struct ami_session *ami, int level)
 {
-	int old_level = ami->debug_level;
+	int old_level = ami ? ami->debug_level : ami_debug_level;
 	if (level < 0 || level > 10) {
 		return -1;
 	}
-	ami->debug_level = level;
+	if (ami) {
+		ami->debug_level = level;
+	} else {
+		ami_debug_level = level;
+	}
 	return old_level;
 }
 
-static int __attribute__ ((format (gnu_printf, 4, 5))) __ami_send(struct ami_session *ami, va_list ap, const char *fmt, const char *prefmt, ...)
+static int __attribute__ ((format (printf, 3, 0))) __attribute__ ((format (printf, 4, 5))) __ami_send(struct ami_session *ami, va_list ap, const char *fmt, const char *prefmt, ...)
```

Comment:
Why is 3,0 needed in addition to 4,5?

Reply:
Both "fmt" and "prefmt" are printf-like format strings.